src/main/java/mchorse/blockbuster/core/transformers/EntityTransformationUtilsTransformer.java

```diff
@@ -0,0 +1,70 @@
+package mchorse.blockbuster.core.transformers;
+
+import mchorse.blockbuster.utils.mclib.coremod.ClassTransformer;
+import net.minecraft.util.EnumFacing;
+import org.objectweb.asm.Opcodes;
+import org.objectweb.asm.tree.*;
+
+import java.util.Iterator;
+
+public class EntityTransformationUtilsTransformer extends ClassTransformer
+{
+    @Override
+    public void process(String s, ClassNode node)
+    {
+        for (MethodNode method : node.methods)
+        {
+            if (method.name.equals("getPrevPrevPosX") && method.desc.equals("(Lnet/minecraft/entity/Entity;)D"))
+            {
+                this.processGetPrevPrevPos(method, "X");
+            }
+            else if (method.name.equals("getPrevPrevPosY") && method.desc.equals("(Lnet/minecraft/entity/Entity;)D"))
+            {
+                this.processGetPrevPrevPos(method, "Y");
+            }
+            else if (method.name.equals("getPrevPrevPosZ") && method.desc.equals("(Lnet/minecraft/entity/Entity;)D"))
+            {
+                this.processGetPrevPrevPos(method, "Z");
+            }
+        }
+
+        System.out.println("BBCoreMod: successfully patched EntityTransformationUtils!");
+    }
+
+    public void processGetPrevPrevPos(MethodNode method, String axis)
+    {
+        Iterator<AbstractInsnNode> it = method.instructions.iterator();
+
+        AbstractInsnNode target = null;
+
+        int index = -1;
+
+        while (it.hasNext())
+        {
+            index++;
+            AbstractInsnNode node = it.next();
+
+            if(node instanceof InsnNode)
+            {
+                if(node.getOpcode() == Opcodes.DCONST_0)
+                {
+                    target = node;
+
+                    break;
+                }
+            }
+        }
+
+        method.instructions.remove(target);
+
+        target =  method.instructions.get(index-1);
+        InsnList list = new InsnList();
+
+        list.add(new VarInsnNode(Opcodes.ALOAD, 0));
+        list.add(new FieldInsnNode(Opcodes.GETFIELD, "net/minecraft/entity/Entity", "prevPrevPos"+axis, "D"));
```

Comment:
Here you need to check for obfuscation and switch between `net/minecraft/entity/Entity` and `vg`, or else it won't work in production.